lib/diagnostic.js

```diff
@@ -38,10 +45,21 @@ const select = function select(array, callback) {
 // Question 5
 // Write a function that returns the first element
 // for which invoking the callback returns a truthy value
-const first = function first(array, callback) {
 
+
+const first = function (array, callback) {
+    if (typeof array[i] !=== 'undefined') {
+        return undefined;
+    }
+
+    for (let i = 0; i < arr.length; i++) {
+        if (i in this. && callback (previousValue,this[i],a)) return this[i];
+    }
+
+    return array[i];
```

Comment:
`i` is not defined outside of the loop. Additionally, if no matching elements can be found, we should probably return `null`.